src/main/scala/devices/debug/SBA.scala

```diff
@@ -101,7 +101,9 @@ object SystemBusAccessModule
       }
     }
 
-    sb2tl.module.io.addrIn := Mux(sb2tl.module.io.rdEn,Cat(SBADDRESSWrData.reverse),Cat(SBADDRESSFieldsReg.reverse))
+    sb2tl.module.io.addrIn := Mux(sb2tl.module.io.rdEn,
+      Cat(SBADDRESSFieldsReg(3), SBADDRESSFieldsReg(2), SBADDRESSFieldsReg(1), SBADDRESSWrData(0)),
```

Comment:
Good catch. But maybe instead of writing all the fields out by hand we could write:
`Cat(Cat(SBADDRESSFieldsReg.drop(1).reverse),SBADDRESSWrData(0))`